### PR TITLE
Jinchao make user invisible to the teams they are on

### PR DESCRIPTION
**Pull request by suigunfjc**
# Description
Front end for: 
image

## Mainly changes explained:
Add a switch at the teams tab on userProfile page. All roles except 'Volunteer' are now invisible to their teammates on the leaderboard by default. 'Owner', 'Administrator', and 'Core Team' can still see them on the leaderboard.

## How to test:
Please test with backend change: [PR300](https://github.com/OneCommunityGlobal/HGNRest/pull/300)
1. Log in as admin and create a team having all classes of users('Volunteer', 'Mentor', 'Manager', 'Core Team', 'Administrator', 'Owner'): 
image
2. Log in to each team member's dashboard and check their leaderboard. Volunteer/mentor/manager should only see volunteer teammates and Core Team/Administrator/Owner will see all teammates.
image

3. Go to Userprofile->Teams, and check if there is a visibility switch for all roles except 'Volunteer'.  Change the status to visible and check if the user can be viewed by all other team members on their leaderboard
image

4. Log in Administrator/Owner to check if they can change the visibility of other users (except 'Volunteer') through userProfile page.

### CODE REVIEW

**Comment by one-community** [comment]
Fantastic!! Please make Manager and Mentor both set to invisible by default too. I'd love this to happen if someone is changed from any other type to a Manager or Mentor too. Is this possible?

**Comment by suigunfjc** [comment]
@one-community All roles except Volunteer are invisible by default. So changing a Volunteer to another role will make them invisible. Though if they are set to visible and changed to other roles, they might need to reset their visibility to invisible manually. If you want to make them invisible in that case, I can raise another PR to fix it. But I think it's better that we test and merge this PR first.

**Comment by one-community** [comment]
Sounds perfect, thank you!

**Comment by AriaYu927** [comment]
Hi @suigunfjc , I've tested your PR by login with each role type and everthing works as expected. Thank you!

One suggestion out of the scope is adding a save button under the "Teams" tab for mentor/core team/manager, so that they don't have to save the change of toggle by switching to "Basic information" tab.
Mentor-team-tab

**Comment by suigunfjc** [comment]
Hi @AriaYu927 , Thank you for the great suggestion, I will address it in another PR after this PR gets merged.

**Review by nicollext — COMMENTED** [review]
Great! I have tested all roles and everything is functioning well.

https://user-images.githubusercontent.com/101591708/227638076-cab8c6cd-2022-4433-9262-d5f3a6abb5c7.MP4

The only case that I was not able to complete the task was when using the browser on a smaller screen. In this case, I am unable to save changes when toggling the user status between visible and invisible. The 'Save Changes' button only works when I make updates to team assignments or deletions."

https://user-images.githubusercontent.com/101591708/227638553-1189da25-9ba5-46fe-9df1-cd0faf355dc3.MP4

**Comment by vitorsaa2k** [comment]
Hello @suigunfjc. While reviewing your PR I've tried to discover what was causing the bug mentioned on @nicollext's review.

I've discovered that you forgot to pass the prop that change the user profile on the component that is shown on smaller screens. Just adding the line on the image could solve the bug.

image

Video after the line has been added:

https://user-images.githubusercontent.com/17365161/227722174-641d0a8e-771a-4b08-8a40-8ba79abc7475.mp4

**Comment by NataliaCavicchioli** [comment]
Hey, @suigunfjc I tested your PR and everything is working fine. 

I added all the roles to the team, logged in each one, and everything is working as intended. Good job!

I just got the same problem with small screens as mentioned by @nicollext.

image

**Comment by NataliaCavicchioli** [comment]
Hey, @suigunfjc I tested your PR and everything is working fine.

I added all the roles to the team, logged in each one, and everything is working as intended. Good job!

With small screen is working well for me too. 

image

**Review by NataliaCavicchioli — DISMISSED** [review]
Looking good. Well done!

**Review by kangjay88 — DISMISSED** [review]
Hi @suigunfjc  , I have tested this PR with the intended roles and it works as expected. I also tested that the visibility option is working as intended. Thank you
Here's an example of an admin's visibility from a volunteer's leaderboard perspective:
Before: 
PR731-2
After: 
PR731-4

**Comment by suigunfjc** [comment]
@vitorsaa2k Great catch, thank you! I will update the branch.

**Comment by suigunfjc** [comment]
Hi guys @vitorsaa2k , @kangjay88 , @NataliaCavicchioli , @nicollext I have fixed the bug that prevented changes from being saved on small screens. Please test the 'save changes' button on a small screen. Thanks!!

**Comment by Jorge-Nerd** [comment]
Hello @suigunfjc  I tested the PR and it work has intended good work
PULL#731

**Comment by one-community** [comment]
Waiting on final checks here and then I'll move this one to final review. Please approve it after your checks. Should be at least 4 approvals here after @vitorsaa2k , @kangjay88 , @NataliaCavicchioli , @nicollext

**Review by AriaYu927 — APPROVED** [review]

**Review by vitorsaa2k — APPROVED** [review]
Tested and it's working as intended!

**Comment by one-community** [comment]
Thank you all, moving to final review.

**Comment by one-community** [comment]
Tested this set. Working well but still have some thing that need fixing in a new PR: 

Mentor: 
No way for this person to save their vis/invis option from own profile
Volunteers can see their tasks over on left (Dashboard Tasks Tab) even when invisible
Made visible after being invisible and they didn’t show up in the Leaderboard for their volunteer team member

Manager and Core Team 
No way for this person to save their vis/invis option from own profile
Volunteers can see their tasks over on left (Dashboard Tasks Tab) even when invisible

Owner and Admins
Volunteers can see their tasks over on left (Dashboard Tasks Tab) even when invisible

See the additional related function requests on the Bugs Doc too.

**Pull request merged**
